libs/designer/src/lib/core/actions/bjsworkflow/settings.ts

```diff
@@ -521,7 +522,10 @@ const getSplitOnValue = (
         // TODO (3727460) - Consume top level required fields when available here.
         const { alias, propertyName, required } = getSplitOnArrayAliasMetadata(manifest.properties.outputs, /* propertyRequired */ true);
         const propertyPath = alias || propertyName;
-        if (propertyPath) {
+        if (propertyPath && (manifest.properties.connectionReference?.referenceKeyFormat !== undefined && manifest.properties.connectionReference?.referenceKeyFormat === ConnectionReferenceKeyFormat.HybridTrigger)) {
+          return `@${Constants.TRIGGER_BODY_OUTPUT}[${convertToStringLiteral(propertyPath)}]`;
+        }
+        else if (propertyPath) {
```

Comment:
revert and make sure you have correct outputs in manifest rather updating like this for a specific trigger.

Reply:
Could you suggest what needs to be the correct output format for adding the SplitOn:
![image](https://github.com/Azure/LogicAppsUX/assets/91710207/c39658d7-3a89-47c0-be7c-b7aa92187592)

Reply:
Why are you adding lines 117-123? Body property is already marked as dynamic properties, so initially you won't have splitonset because outputs do not contain an array to set as spliton options.

Once you have dynamic properties call made, please debug to check what does the outputs in updateoutputsandtokens contain.
Please debug the code to see where it is going wrong